File: src/main/java/com/pigutu/app/mapper/UserDao.java
package com.pigutu.app.mapper;

import com.pigutu.app.entity.UserEntity;
import com.pigutu.app.mapper.mybatis.BaseDao;
import com.pigutu.app.mapper.mybatis.DBMeta;
import org.apache.ibatis.annotations.Mapper;
import org.springframework.stereotype.Repository;

@Mapper
@Repository // 不加也可以
@DBMeta(table = UserDao.TABLE)
public interface UserDao extends BaseDao<UserEntity> {
    String TABLE = "user";
}

File: src/main/java/com/pigutu/app/xforum/dao/UserDao.kt
package com.pigutu.app.xforum.dao

import com.pigutu.app.mapper.CommentDao
import com.pigutu.app.mapper.mybatis.BaseDao
import com.pigutu.app.mapper.mybatis.DBMeta
import com.xforum.library.model.User
import org.apache.ibatis.annotations.Mapper
import org.springframework.stereotype.Repository

@Mapper
@Repository
@DBMeta(table = "x_user")
interface UserDao: BaseDao<User> {

}